WooCommerce/TodayStatsWidget/TodayViewController.swift

```diff
@@ -0,0 +1,366 @@
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as TodayStatsTableViewCell where row == .todayStats:
+            configureTodayStats(cell: cell)
+        case let cell as BasicTableViewCell where row == .errorMessage:
+            configureGenericMessage(cell: cell)
+        default:
+            fatalError()
+            break
+        }
+    }
+
+    func configureTodayStats(cell: TodayStatsTableViewCell) {
+        if let siteURL = site?.url {
+            cell.configure(visitors: statsData?.totalVisitorsReadable(),
+                           orders: statsData?.totalOrdersReadable(),
+                           revenue: statsData?.totalRevenueReadable(),
+                           site: siteURL)
+        }
+    }
+
+    func configureGenericMessage(cell: BasicTableViewCell) {
+        cell.textLabel?.numberOfLines = 0
+        cell.backgroundColor = .clear
+
+        switch error {
+        case .statsV4NotAvailable:
+            cell.textLabel?.text = LocalizedText.analyticsNotAvailable
+        case .errorFetchingStats:
+            cell.textLabel?.text = LocalizedText.errorFetchingStats
+        case .userNotLogged:
+            cell.textLabel?.text = LocalizedText.missingCredentials
+        case .none:
+            cell.textLabel?.text = LocalizedText.genericError
+        }
+    }
+}
+
+// MARK: - Private Types
+//
+private extension TodayViewController {
+
+    struct Section {
+        let rows: [Row]
+    }
+
+    enum Row: CaseIterable {
+        case todayStats
+        case errorMessage
+
+        var type: UITableViewCell.Type {
+            switch self {
+            case .todayStats:
+                return TodayStatsTableViewCell.self
+            case .errorMessage:
+                return BasicTableViewCell.self
+            }
+        }
+
+        var reuseIdentifier: String {
+            return type.reuseIdentifier
+        }
+    }
+}
+
+// MARK: - Constants
+//
+private extension TodayViewController {
+
+    enum LocalizedText {
+        static let analyticsNotAvailable = NSLocalizedString(
+            "Store analytics not available! Please upgrade to the latest version of WooCommerce to view your store analytics.",
+            comment: "Store analytics error in Today Stats Widget")
+        static let errorFetchingStats = NSLocalizedString("There was an error trying to fetch stats.",
+                                                          comment: "Error while fetching stats in Today Stats Widget")
+        static let missingCredentials = NSLocalizedString("Please log in to the WooCommerce app to add a widget.",
+                                                          comment: "Login error in Today Stats Widget")
+        static let genericError = NSLocalizedString("Something goes wrong.", comment: "Generic error in Today Stats Widget")
+    }
+
+    enum Errors {
```

Comment:
nit: it's a bit odd seeing the error property like `error: Errors`, maybe this could be like `enum StatsWidgetError: Error`/`enum WidgetError: Error`?